### DISCUSSION TITLE
Make GeometryNode supported on ARSceneView

### DISCUSSION THREAD

**Discussion by andrewmo2014** [sceneview/sceneview#343] [Ideas]
Are creating custom geometries currently supported for ARSceneView? With the latest architecture changes made to Sceneview, I would like to confirm if GeometryNodes can work for ARSceneView as well. It would be great to have a sample showing how to customize GeometryNodes too. For example, being able to generate custom geometries at runtime similar to the original Sceneform drawing sample located at [sceneform-samples/drawing at master · googlesamples/sceneform-samples (github.com)](https://github.com/googlesamples/sceneform-samples/tree/master/drawing). What would be needed to accomplish this?

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, all geometry nodes work in both `SceneView` and `ARSceneView`:

```kotlin
ARSceneView {
    AnchorNode(anchor = anchor) {
        CubeNode(materialInstance = material, size = Size(0.2f))
        SphereNode(materialInstance = material, radius = 0.1f)
        CylinderNode(materialInstance = material, radius = 0.1f, height = 0.3f)
        MeshNode(materialInstance = material, vertices = ..., indices = ...)
    }
}
```

There's no longer a distinction — all node types are composables that work in any scene.
